src/server/frames.ts

```diff
@@ -263,20 +263,21 @@ export class FrameManager {
         route.continue();
       return;
     }
+    this._page._browserContext.emit(BrowserContext.Events.Request, request);
     this._page._requestStarted(request);
   }
 
   requestReceivedResponse(response: network.Response) {
     if (response.request()._isFavicon)
       return;
     this._responses.push(response);
-    this._page.emit(Page.Events.Response, response);
+    this._page._browserContext.emit(BrowserContext.Events.Response, response);
```

Comment:
I think the fact that you've dropped the `Page.Events.Response` and similar ones might break some internal listeners, e.g. `HarTracer`. I hope that tests should point that 😄

Reply:
Do we need to emit events on page as well? Since we already do emit it for the consuming part on `client/browserclient.ts`?

Reply:
Ahh i get it, its for the server side stuff? I brought them back, but we can move to BrowserContext.Request in another PR to make sure we dont miss our requests with HAR when popups are involved.

Reply:
> ... but we can move to BrowserContext.Request in another PR to make sure we dont miss our requests with HAR when popups are involved.

Nice catch!